lib/selenium.pm

```diff
@@ -0,0 +1,190 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017, 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Support for Selenium
+# Maintainer: Vladimir Nadvornik <nadvornik@suse.com>
+
+package selenium;
+use 5.018;
+
+use base 'Exporter';
+use Exporter;
+
+our @EXPORT = qw(
+  add_chromium_repos
+  install_chromium
+  enable_selenium_port
+  selenium_driver
+  wait_for_page_to_load
+  wait_for_link
+  wait_for_text
+  wait_for_xpath
+  select_input
+);
+
+use testapi;
+use utils 'zypper_call';
+
+use Selenium::Remote::Driver;
+use Selenium::Chrome;
+use Selenium::Waiter qw/wait_until/;
+use Selenium::Remote::WDKeys;
+
+my $port = 4444;
+
+=head1 openQA - selenium webdriver support
+
+This Selenium module exports subroutines helpers to integrate C<Selenium::Remote::Driver>
+into openQA tests
+
+Note: Selenium module works only when openvswitch networking is used!
+
+Chromedriver and chromium are installed on SUT, firewall is enabled. OpenQA worker then initiate
+connection to SUT and returns C<Selenium::Remote::Driver> to be used in tests.
+
+Usage:
+
+  use selenium;
+  add_chromium_repos;   # for SLES12
+
+  install_chromium;
+  enable_selenium_port;
+
+  my $driver = selenium_driver;
+  ...
+=cut
+sub add_chromium_repos {
+    my $ret = zypper_call("se chromedriver", exitcode => [0, 104]);
+    if ($ret == 104) {
+        zypper_call(
+            '--gpg-auto-import-keys ar -fc http://download.opensuse.org/repositories/openSUSE:/Backports:/SLE-12/standard/openSUSE:Backports:SLE-12.repo');
+        zypper_call(
+'--gpg-auto-import-keys ar -fc http://download.opensuse.org/repositories/openSUSE:/Backports:/SLE-12-SP1/standard/openSUSE:Backports:SLE-12-SP1.repo'
+        );
+        zypper_call(
+'--gpg-auto-import-keys ar -fc http://download.opensuse.org/repositories/openSUSE:/Backports:/SLE-12-SP2/standard/openSUSE:Backports:SLE-12-SP2.repo'
+        );
+        zypper_call('--gpg-auto-import-keys ar -fc http://smt.suse.cz/repo/SUSE/Products/SLE-SDK/12-SP2/x86_64/product/ sdk');
+        zypper_call('--gpg-auto-import-keys ref');
+    }
+}
+
+sub install_chromium {
+    zypper_call('in chromium chromedriver');
+    script_run("ln -s /usr/lib64/chromium/chromedriver /usr/bin/chromedriver");
+}
+
+sub enable_selenium_port {
+    assert_script_run("sed -i -e 's|FW_SERVICES_EXT_TCP=\"\\(.*\\)\"|FW_SERVICES_EXT_TCP=\"\\1 $port\"|' /etc/sysconfig/SuSEfirewall2");
+    assert_script_run("rcSuSEfirewall2 restart");
```

Comment:
This thing will be broken on SLE15, Leap 15 and TW as we have firewalld there now

Reply:
Ok, this will take a while. I never worker with firewalld, need to get up to speed yet.

Reply:
I guess it's also fine to simply stop firewall using [this method here](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/opensusebasetest.pm#L497) to get service name and then call
```perl
systemctl 'stop ' . $self->firewall
```